docs/configuration.md

```diff
@@ -0,0 +1,158 @@
+## Configuration
+
+All configuration should go into `config/initializers/wallaby.rb`. See the following sections for the configuration that you are looking for.
+
+- [Authentication](#authentication)
+- [Logging Out](#logging-out)
+- [Authorization](#authorization)
+- [Models](#models)
+- [Metadata](#metadata)
+- [Pagination](#pagination)
+- [Features](#features)
+
+### Authentication
+
+In order to do authentication, Wallaby follows the common practice to execute `authenticate_user!` in before action callbacks and use `current_user` to return the user object. Therefore, there are two ways to set up authentication:
+
+1. Simply tell Wallaby which controller to inherit from. The controller should have `authenticate_user!` and `current_user` implemented:
+
+    ```ruby
+    #!config/initializers/wallaby.rb
+    Wallaby.config do |config|
+      config.base_controller = MySecurityController
+    end
+    ```
+
+    Once this is set up, Wallaby will automatically pick up these two authentication methods mentioned above (which are compatible with Devise), along with all functionalities including application helpers, before_action, etc.
+
+2. Authentication can be customized by configuring the `authenticate` and `current_user` (optional: `email_method`) options as per the example below:
+
+    ```ruby
+    #!config/initializers/wallaby.rb
+    Wallaby.config do |config|
+      config.security.authenticate do
+        # controller methods can be used inside this block
+        authenticate_or_request_with_http_basic do |username, password|
+          username == 'too_simple' && password == 'too_naive'
+        end
+      end
+
+      config.security.current_user do
+        # controller methods can be used inside this block
+        Class.new do
+          # email field is for gravator profile image
+          def email
+            'user@example.com'
+          end
+        end.new
+      end
+
+      # This will be used to retrive the email from current_user above to show a gravatar image.
+      config.security.email_method = :email
+    end
+    ```
+
+### Logging Out
+
+Wallaby doesn't do log-out function. However, it provides two options to send user to the logout route:
```

Comment:
It may be good to consistently use the sam term `log-out` or alternatively `logout`.